pandas/io/parsers.py

```diff
@@ -656,7 +658,13 @@ def _get_options_with_defaults(self, engine):
         options = {}
 
         for argname, default in compat.iteritems(_parser_defaults):
-            options[argname] = kwds.get(argname, default)
+            value = kwds.get(argname, default)
+
+            if argname == 'mangle_dupe_cols' and value != default:
+                raise ValueError('Setting mangle_dupe_cols=False is '
+                                 'not supported yet')
```

Comment:
Yeah, so if `mangle_dupe_cols=True`, there is no duplicate data issue.  However, if `mangle_dupe_cols=False`, the duplicate data issue CAN arise if either the header or `names` contain duplicates.